cogfiles/birthdays.py

```diff
@@ -67,15 +85,15 @@ async def birthdays(self, interaction: discord.Interaction):
         # Add the dates that have already happened after the dates that haven't
         # This gives a list of upcoming birthdays in sorted order, including some dates from next year after this year
         now = datetime.datetime.now().date()
-        next_birthdays = {user: birthday.replace(year=now.year) for user, birthday in self.birthdays[interaction.guild.id].items()}
-        sorted_birthdays = {user: birthday for user, birthday in sorted(next_birthdays.items(), key=lambda item: item[1])}
-        sorted_birthdays = {user: birthday for user, birthday in sorted_birthdays.items() if now < birthday} | {user: birthday.replace(year=now.year+1) for user, birthday in sorted_birthdays.items() if now >= birthday}
+        next_birthdays = {Birthday(birthday.user, birthday.date.replace(year=now.year)) for birthday in self.birthdays[interaction.guild.id]}        
+        sorted_birthdays = sorted(next_birthdays, key=lambda birthday: birthday)

```

Comment:
I don't think the lambda function you've passed into `key` here is actually doing anything. Unless I'm mistaken, you should be able to simply call `sorted(next_birthdays)` in this case.

Reply:
Oops I forgot to change the lambda expression. It should read `lambda birthday: birthday.date`, I'll fix that